returnn/datasets/basic.py

```diff
@@ -1410,12 +1439,14 @@ def init_dataset(
     default_kwargs: Optional[Dict[str, Any]] = None,
     *,
     parent_dataset: Optional[Dataset] = None,
+    forward_sharding_config: bool = False,
```

Comment:
I don't understand this. Why is such an argument needed? It should just always be forwarded.

And even if there is a case when this is not wanted, you would not need such an option here. You can simply put `num_shards=0` to the `kwargs` or `extra_kwargs` to explicitly disable it.